src/Hydra/Serializer/DocumentationNormalizer.php

```diff
@@ -185,7 +186,9 @@ private function getHydraProperties(string $resourceClass, ResourceMetadata $res
             }
 
             if ($this->nameConverter) {
-                $propertyName = $this->nameConverter->normalize($propertyName);
+                $propertyName = $this->nameConverter instanceof AdvancedNameConverterInterface
```

Comment:
You can drop the test, just pass all arguments: they will be ignored for implementations not implementing `AdvancedNameConverterInterface`.
Also, can you pass the the format and the context?

Same for the other one